### DISCUSSION TITLE
GUI for WinGet

### DISCUSSION THREAD

**Discussion by PhilippBaumi** [microsoft/winget-cli#733] [Ideas]
My suggestions for improvements: 
1. Make Microsoft Store or App Installer the GUI for WinGet 
2. Distribute all programs (Win32 and Store Apps) as well as Windows 10 updates (Windows 10 versions, cumulative updates, driver,...) via WinGet.  So, in my opinion, WinGet should combine the installation of the operating system and programs in a single tool. 

Then WinGet would be the one-stop shop to set up the computer and update it. Then you would no longer have to use WinGet (for programs and apps from the Store) and Windows Updates (for Windows Updates and Drivers), but would only have to use WinGet and you would update everything (as it works in Linux too).

**Comment by alex47** [comment]
I agree with point 1.
This is exactly what the Windows Store should be, instead of all the low quality programs full with advertisements.

**Comment by denelon** [comment]
I'm mentioning:
* https://github.com/microsoft/winget-cli/issues/820

To continue the discussion related to a GUI for WinGet.

For now, the Microsoft Store UI is not in scope. I just didn't want to "clobber" or ignore this existing discussion since the title is so accurate to what we're beginning to look at.

**Reply by riverar** [reply]
I don't have a strong opinion yet but I can't think of any reason I'd use/need a WinGet GUI. Feels like a distraction from more important winget issues/tasks.

**Reply by marticliment** [reply]
I agree. Winget still has lots of possible improvements/enhacements to do, and given the amount of third-party free options that there are (as listed [here](https://github.com/microsoft/winget-cli/issues/820#issuecomment-1398709391)), I don't see an urgent need for an official GUI to appear.

**Reply by marticliment** [reply]
However, great improvements could be made to Winget that would make GUIs more integrable, such as providing native support for icons and/or other information

**Comment by denelon** [comment]
@marticliment icons are certainly one of the considerations we're looking into. When we run our tests on packages during validation, they are installed on a machine which gives us access to the icon. We're thinking about how we would leverage that and make it available via some API for use in a GUI.

**Reply by marticliment** [reply]
that would be ideal, and maybe also the ability for developers to share screenshots (optional) on their manifests, because this would make websites like winstall or winget.run way better

**Reply by DrewNaylor** [reply]
Screenshot links in manifests would be a really good idea, because then it would be less important for each GUI to implement their own way of getting screenshots (they could probably still augment the available screenshots if a particular manifest doesn't list any, but that would be a fallback or maybe even an "extra screenshots" feature or something, in case the others are unreachable or more are needed or something).

**Comment by denelon** [comment]
@Stanzilla, 
The Microsoft Store UI has been designed with consumers as the primary audience. We are meeting with them on a regular basis. There is still heavy lifting to be done on several fronts with the WinGet integration there. 

We see several benefits with building a WinGet GUI and getting some lessons learned in an iterative fashion. The Microsoft Store has explicit agreements between publishers/ISVs (Independent Software Vendors) and Microsoft. The community repository has no such formal agreements in place.

I would like to innovate & incubate new ideas and get customer feedback. I'll use PowerToys as an example where we can rapidly innovate and then for those things that make sense, we can bring them natively into Windows. 

There are many things we do with WinGet to learn about customer preferences and iterate before things are brought into the Microsoft Store. The support for Win32 applications in the store was after we had built up some of the learnings with WinGet.

**Reply by Stanzilla** [reply]
While I do agree with the principle of incubating in general, in this case it could backfire and alienate developers more. There are already so many ways to install apps on Windows and if you tell people that some stuff is only on the store, some only on winget, some only with their own installers, Not saying this is bad, just that it can cause confusion.

But you mentioned PowerToys...maybe this could/should be part of PowerToys then? I just want to avoid the case of people having to "get" yet another thing.

**Reply by denelon** [reply]
It was actually brought up there:
* https://github.com/microsoft/PowerToys/issues/22925

**Reply by Stanzilla** [reply]
Yeah I think I like that idea, they also have people with tons of experience working on modern Windows UI, would be a good team collab project?
